nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,440 @@
+            except StopIteration:
+                # Empty tractogram
+                header[Field.NB_STREAMLINES] = 0
+                # Overwrite header with updated one.
+                f.seek(beginning, os.SEEK_SET)
+                self._write_header(f, header)
+
+                # Add the EOF_DELIMITER.
+                f.write(asbytes(self.EOF_DELIMITER.tostring()))
+                return
+
+            data_for_streamline = first_item.data_for_streamline
+            if len(data_for_streamline) > 0:
+                keys = ", ".join(data_for_streamline.keys())
+                msg = ("TCK format does not support saving additional data"
+                       " alongside streamlines. Dropping: {}".format(keys))
+                warnings.warn(msg, DataWarning)
+
+            data_for_points = first_item.data_for_points
+            if len(data_for_points) > 0:
+                keys = ", ".join(data_for_points.keys())
+                msg = ("TCK format does not support saving additional data"
+                       " alongside points. Dropping: {}".format(keys))
+                warnings.warn(msg, DataWarning)
+
+            # Make sure streamlines are in rasmm.
+            tractogram = self.tractogram.to_world(lazy=True)
+
+            for t in tractogram:
+                s = t.streamline.astype(dtype)
+                data = np.r_[s, self.FIBER_DELIMITER]
+                f.write(asbytes(data.tostring()))
+                nb_streamlines += 1
+
+            header[Field.NB_STREAMLINES] = nb_streamlines
+
+            # Add the EOF_DELIMITER.
+            f.write(asbytes(self.EOF_DELIMITER.tostring()))
+
+            # Overwrite header with updated one.
+            f.seek(beginning, os.SEEK_SET)
+            self._write_header(f, header)
+
+    @staticmethod
+    def _write_header(fileobj, header):
+        """ Write TCK header to file-like object.
+
+        Parameters
+        ----------
+        fileobj : file-like object
+            An open file-like object pointing to TCK file (and ready to read
+            from the beginning of the TCK header).
+        """
+        # Fields to exclude
+        exclude = [Field.MAGIC_NUMBER,  # Handled separately.
+                   Field.NB_STREAMLINES,  # Handled separately.
+                   Field.ENDIANNESS,  # Handled separately.
+                   Field.VOXEL_TO_RASMM,  # Streamlines are always in RAS+ mm.
+                   "count", "datatype", "file"]  # Fields being replaced.
+
+        lines = []
+        lines.append(header[Field.MAGIC_NUMBER])
+        lines.append("count: {0:010}".format(header[Field.NB_STREAMLINES]))
+        lines.append("datatype: Float32LE")  # Always Float32LE.
+        lines.extend(["{0}: {1}".format(k, v)
+                      for k, v in header.items() if k not in exclude and not k.startswith("_")])
+        lines.append("file: . ")  # Manually add this last field.
+        out = "\n".join((asstr(line).replace('\n', '\t') for line in lines))
```

Comment:
Bear in mind that generally nobody reads warnings.  I guess, raise an error if you can't work out what the user meant to do, or if they may be using the API incorrectly, but just warn if there are minor defects in the input.